### DISCUSSION TITLE
在G2 5.1 中如何获取 axis 的所有值，也就是axis的label的集合

### DISCUSSION THREAD

**Discussion by PDieE** [antvis/G2#5629] [Q&A] [answered]
- **G2 Version**: 5.1
- **Platform**:
- **Mini Showcase(like screenshots)**:
- **CodePen Link**:

我想在图表中绘制一些和axis的label有关的自定义图形，但是我找不到如何获取相关数据的方法

**Comment by pearmini** [comment]
是想获取 label 的文本数据？

**Comment by PDieE** [comment]
> 是想获取 label 的文本数据？

最终目的是想获取axis的label的文本数据集合

**Comment by pearmini** [comment]
```js
import { LABEL_CLASS_NAME } from '@antv/g2';

chart.render(() => {
  const { canvas } = chart.getContext();
  const document = canvas.document;
  const labels = document.getElementsByClassName(LABEL_CLASS_NAME);
  // ...
});
```

**Comment by PDieE** [comment]
> ```js
> import { LABEL_CLASS_NAME } from '@antv/g2';
> 
> chart.render(() => {
>   const { canvas } = chart.getContext();
>   const document = canvas.document;
>   const labels = document.getElementsByClassName(LABEL_CLASS_NAME);
>   // ...
> });
> ```

试了一下，一个是 render没有这个回调
然后我换到其他地方进行调用以后 获取到的labels 是个空数组，且我两个轴的都没关闭

![image](https://github.com/antvis/G2/assets/14084550/8a1f552b-6358-4f06-9ed5-a9e31cc8b078)

**Reply by pearmini** [reply, marked as answer]
写错了：

```js
chart.render().then(callback)
```

我以为你说的 label，axis label 可以可以先获得 axis（className 就是 axis），然后打印出来，看看 children 里面 label 的 className

**Reply by PDieE** [reply]
谢谢，之前我也是用的这种办法，主要是想有没有什么更简单的办法
